explorer/frontend/src/Lib/Waypoints/Waypoints.purs

```diff
@@ -42,4 +43,8 @@ waypoint elemId handler = runEffFn3 waypointImpl elemId handler defaultWaypointO
 waypoint' :: forall eff. ElementId -> (WaypointHandler eff) -> WaypointOffset -> Eff (waypoint :: WAYPOINT | eff) Waypoint
 waypoint' = runEffFn3 waypointImpl
 
-foreign import destroy :: forall eff. Fn1 Waypoint (Eff (waypoint :: WAYPOINT | eff) Unit)
+foreign import destroyImpl :: forall eff. EffFn1 (waypoint :: WAYPOINT | eff) Waypoint Waypoint
```

Comment:
@sectore I don't know the details, but shuldn't destroy return `Unit` instead of `Waypoint`.  Something that is destroy shouldn't return reference to it (in general - it makes a little sense). Did you mean:
```haskell
foreign import destroyImpl :: forall eff. EffFn1 (waypoint :: WAYPOINT | eff) Waypoint Unit
```

?

Reply:
`destroy` returns an instance of `Waypoint` - see http://imakewebthings.com/waypoints/api/destroy/

Reply:
Waypoint is used to animate the size of the header of Dashboard depending on current scroll position.

Reply:
hm, in that case I believe https://github.com/input-output-hk/cardano-sl/pull/2377#discussion_r163841639 should be sufficient. Right?

Reply:
Can you try just with this modification (just to check it out):
```
foreign import destroy :: forall eff. Fn1 Waypoint (Eff (waypoint :: WAYPOINT | eff) Waypoint)
```

ie, that it returns Waypoint instead of Unit. Because that should be the only difference between the old and new implementation.

Reply:
Just tried it ^ out as you suggested. It still throws `TypeError: Waypoints.destroy(...) is not a function`. The only way to fix this seems to be wrapping that foreign function.

Reply:
Strange. Ok, lets then proceed with it.

Reply:
Ah no, it makes sense. In fact I was always using EffFn modifiers and they worked (don't have much exp with Fn1 modifiers and should they be unwrapped somhow as well as with EffFn and runEffFn). Approving